### DISCUSSION TITLE
Задание 7

### DISCUSSION THREAD

**Discussion by eprilutskaya** [andyacedoc/onceupon#25] [General]
https://github.com/andyacedoc/onceupon/blob/cfd0220cb293a2113028adfd9d66530b08d9a02a/AndyModule/Cron/SendBlackListMessage.php#L77
В foreach можно передавать коллекцию, те получить ее используя Factory:
передать в __construct \Amasty\AndyModule\Model\ResourceModel\Blacklist\CollectionFactory $collectionFactory
$blackListCollection = $collectionFactory->create();
foreach ($blackListCollection as $blackList) {
...
$blackList->setData('email_body', $bodyMessage);
...
}

Для сохранения используем ресурсную модель \Amasty\AndyModule\Model\ResourceModel\Blacklist
передать в __construct \Amasty\AndyModule\Model\ResourceModel\Blacklist $blacklistResource
$blacklistResource->save($blackList)

Колонку email_body нужно добавить используя UpgradeSchema, по коду сейчас колонка не добавлена, а значит текст не сохранится

**Comment by andyacedoc** [comment]
Да, не делал обновление с UpgradeSchema. Для ускорения процесса я добавил колонку вручную через phpMyAdmin.

**Reply by eprilutskaya** [reply]
Как раз это нужно сделать, чтобы показать, что разобрались, как работают скрипты и что нужно для обновления таблицы
